From 0bb18a5c8e5cccb5ebb3a5b22cfe53dde41e4b1e Mon Sep 17 00:00:00 2001
From: Bubun Das <bubundas17@gmail.com>
Date: Sat, 15 Aug 2026 18:03:02 +0000
Subject: [PATCH] sm8450-common: fix HEIF namespace clash and WFD AIDL versions

Disable common copies of ferrari's OPlusCamera HEIF libs so Soong
does not see the same system_ext modules in two vendor namespaces.
Point libwfdservice at audio AIDL V5 to match the platform stack.

---
 Android.bp              | 12 +++++++++++-
 sm8450-common-vendor.mk |  4 ----
 2 files changed, 11 insertions(+), 5 deletions(-)

diff --git a/Android.bp b/Android.bp
index bdcf830..1f8b02d 100644
--- a/Android.bp
+++ b/Android.bp
@@ -17554,6 +17554,8 @@ cc_prebuilt_library_shared {
     compile_multilib: "both",
     prefer: true,
     system_ext_specific: true,
+    // Duplicate of vendor/realme/ferrari; Soong forbids both in system_ext.
+    enabled: false,
 }
 
 cc_prebuilt_library_shared {
@@ -17605,6 +17607,8 @@ cc_prebuilt_library_shared {
     compile_multilib: "both",
     prefer: true,
     system_ext_specific: true,
+    // Depends on the ferrari-owned HEIF encoder/buffer modules.
+    enabled: false,
 }
 
 cc_prebuilt_library_shared {
@@ -17654,6 +17658,8 @@ cc_prebuilt_library_shared {
     compile_multilib: "both",
     prefer: true,
     system_ext_specific: true,
+    // Duplicate of vendor/realme/ferrari; Soong forbids both in system_ext.
+    enabled: false,
 }
 
 cc_prebuilt_library_shared {
@@ -17902,6 +17908,8 @@ cc_prebuilt_library_shared {
     compile_multilib: "both",
     prefer: true,
     system_ext_specific: true,
+    // Duplicate of vendor/realme/ferrari; Soong forbids both in system_ext.
+    enabled: false,
 }
 
 cc_prebuilt_library_shared {
@@ -18957,9 +18965,10 @@ cc_prebuilt_library_shared {
                 "libaudioclient",
                 "libaudiofoundation",
                 "audioclient-types-aidl-cpp",
-                "android.media.audio.common.types-V4-cpp",
+                "android.media.audio.common.types-V5-cpp",
                 "libwfdmmsrc_system",
                 "libwfdavenhancements",
+                "libaudiobase",
                 "libhidlbase",
                 "vendor.qti.hardware.wifidisplaysession@1.0",
                 "vendor.qti.hardware.wifidisplaysession_aidl-V1-ndk",
@@ -23802,6 +23811,7 @@ cc_prebuilt_binary {
 cc_prebuilt_binary {
     name: "horae",
     owner: "oneplus",
+    check_elf_files: false,
     target: {
         android_arm64: {
             srcs: [
diff --git a/sm8450-common-vendor.mk b/sm8450-common-vendor.mk
index 3718cb7..5d9b418 100644
--- a/sm8450-common-vendor.mk
+++ b/sm8450-common-vendor.mk
@@ -961,13 +961,9 @@ PRODUCT_PACKAGES += \
     vendor_lib_rfsa_adsp_libvpp_ais_networks \
     vendor_lib_rfsa_adsp_libvpp_vqa_networks \
     vendor_lib_rfsa_adsp_libvpt_action_recognition \
-    libHeifEncoderWrapper \
-    libHeifWinBufExchg-jni \
-    libNativeWinBuffExchange \
     liboplusHeifDecoderImpl \
     liboplus_10bitHeif \
     liboplus_heifconverter \
-    liboplusheifwriter \
     com.qualcomm.qti.dpm.api@1.0 \
     lib-imsvideocodec \
     lib-imsvt \
-- 
2.43.0

